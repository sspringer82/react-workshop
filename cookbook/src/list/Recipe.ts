type Recipe = {
  id: number;
  title: string;
  ingredients: string[];
  steps: string[];
};

export default Recipe;
