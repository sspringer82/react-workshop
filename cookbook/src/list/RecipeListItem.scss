.ListRecipeListItem {
  width: 400px;
  border: 1px solid black;
  margin: 3px 10px;
  padding: 5px;
  position: relative;
  box-shadow: 0px 2px 2px 0px lightgrey;

  button {
    position: absolute;
    right: 10px;
  }
}
