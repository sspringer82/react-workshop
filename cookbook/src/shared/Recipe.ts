type Recipe = {
  id: number;
  title: string;
  ingredients: string[];
  steps: string[];
};

export default Recipe;

export type InputRecipe = Omit<Recipe, 'id'> & { id?: number };
