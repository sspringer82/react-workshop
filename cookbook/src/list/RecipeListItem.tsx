import { ReactElement } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import Recipe from './Recipe';
import './RecipeListItem.scss';

type Props = {
  recipe: Recipe;
  onDelete: (id: number) => void;
};

const RecipeListItem = ({ recipe, onDelete }: Props): ReactElement => {
  const navigate = useNavigate();
  return (
    <div className="ListRecipeListItem">
      {recipe.title}
      <button onClick={() => onDelete(recipe.id)}>lÃ¶schen</button>
      {/* <Link to={`/detail/${recipe.id}`}>details</Link> */}
      <button onClick={() => navigate(`/detail/${recipe.id}`)}>details</button>
    </div>
  );
};

export default RecipeListItem;
