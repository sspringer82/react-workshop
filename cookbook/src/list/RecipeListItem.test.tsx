import { act, fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Recipe from '../shared/Recipe';
import RecipeListItem from './RecipeListItem';

describe('RecipeListItem', () => {
  // arrange
  const recipe: Recipe = {
    id: 1,
    title: 'Nudelsuppe',
    ingredients: [],
    steps: [],
  };
  it('should render', () => {
    // act
    act(() => {
      render(
        <MemoryRouter>
          <RecipeListItem recipe={recipe} onDelete={jest.fn()} />
        </MemoryRouter>,
      );
    });

    // assert
    expect(screen.getByTestId('title').textContent).toBe('Nudelsuppe');
  });

  it('should trigger onDelete when the delete button is clicked', () => {
    const onDeleteSpy = jest.fn();

    act(() => {
      render(
        <MemoryRouter>
          <RecipeListItem recipe={recipe} onDelete={onDeleteSpy} />
        </MemoryRouter>,
      );
    });

    fireEvent.click(screen.getByTestId('delete-button'));

    expect(onDeleteSpy).toHaveBeenCalledTimes(1);
    expect(onDeleteSpy).toHaveBeenCalledWith(1);
  });

  it('should do a snapshot', () => {
    // act
    const { asFragment } = render(
      <MemoryRouter>
        <RecipeListItem recipe={recipe} onDelete={jest.fn()} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
