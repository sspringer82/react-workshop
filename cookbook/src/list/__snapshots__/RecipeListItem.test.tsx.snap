// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListItem should do a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ListRecipeListItem"
  >
    <span
      data-testid="title"
    >
      Nudelsuppe
    </span>
    <button
      data-testid="delete-button"
    >
      l√∂schen
    </button>
    <button>
      details
    </button>
  </div>
</DocumentFragment>
`;
